sql/core/src/main/scala/org/apache/spark/sql/DataFrameWriter.scala

```diff
@@ -130,6 +131,16 @@ final class DataFrameWriter private[sql](df: DataFrame) {
   }
 
   /**
+   * Specify the compression codec when saving it on hdfs
+   *
+   * @since 1.7.0
+   */
+  def compress(codec: Class[_ <: CompressionCodec]): DataFrameWriter = {
+    this.extraOptions += ("compression.codec" -> codec.getCanonicalName)
```

Comment:
The amount of code changes should be small, so we do not need this additional interface.
If we add an interface for each additional option, the number of interfaces blows up.